### PR TITLE
Fix vertical alignment of notice component

### LINKED ISSUE #75270: Vertically align icon on Notice component to the line-height of the first line of the text.
As I understand this Notice component can support multi-line messages, it would be nice if the icon could be vertically-aligned to the line-height of the first line of the text.
image

_Originally posted by @yansern in https://github.com/Automattic/wp-calypso/issues/75243#issuecomment-1495729383_

### PR DESCRIPTION
**Pull request by TimBroddin**
Closes #75270

## Proposed Changes

This PR adds a negative margin to the content of the Notice component.

| Before | After |
|--------|------|
| CleanShot 2023-04-04 at 14 20 28@2x | CleanShot 2023-04-04 at 14 21 14@2x |

There was a bug in the previous implementation, it was using CSS variables that were unavailable:

CleanShot 2023-04-04 at 13 42 49@2x

That's been fixed as well, by using the SCSS variables. I replaced `font-body-small` with `font-body`, as that was the size that was used before, and `font-body-small` didn't look right.

## Testing Instructions

1. Apply this PR and make sure you have a cookie subkey set
2. Navigate to http://calypso.localhost:3000/subscriptions/sites
3. Delete all your subscriptions
4. Verify 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/subscriptions/notice/style.scss [hunk 1]
```diff
@@ -10,12 +10,13 @@
...
-		background: var(--studio-gray-0);
+		background: $studio-gray-0;
 
 		&-content {
-			font-size: var(--font-body-small);
-			line-height: var(--font-title-small);
+			font-size: $font-body;
+			line-height: $font-title-small;
 			letter-spacing: -0.15px;
+			margin-top: -2px;
```
Comment (phcp) [inline client/landing/subscriptions/notice/style.scss]: On my tests, the icon still seems to be misaligned with the first line.
Screenshot 2023-04-04 at 10 00 24
If we set the icon container to be the same size as the line height and remove the margin it seems to work as expected.
- Removed `margin-top: -2px;`
- Added the CSS below:
```css
		&-icon {
			height: 20px;
		}
```
image

Comment (TimBroddin) [reply client/landing/subscriptions/notice/style.scss]: Nice! Thanks!

Comment (TimBroddin) [reply client/landing/subscriptions/notice/style.scss]: Looking closer at this, I think @yansern means aligning the top of the Y with the top of the icon.

Maybe 2px is not enough?

Comment (yansern) [reply client/landing/subscriptions/notice/style.scss]: Well 2px was correct before the font size change I think! But the point of is just to make sure it is vertically aligned middle with the line-height of the first line of the sentence. I'm cool with any approach to fix it.

Comment (TimBroddin) [reply client/landing/subscriptions/notice/style.scss]: The font size seems the same in the before/after screenshots. I did some calculations earlier and I think if you want to align the top of icon with top of letters, it's closer to 3.5px. 

But vertical alignment is hard, and I think @phcp's solution looks way more natural than mine (ours 😁). Aligning the top of the text (capital Y) with the top of the icon feels cramped to me.

Comment (yansern) [reply client/landing/subscriptions/notice/style.scss]: > The font size seems the same in the before/after screenshots. 

Ah okay, then it must be the difference I'm seeing between a non-retina display and a retina screenshot.

> But vertical alignment is hard, and I think @phcp's solution looks way more natural than mine (ours 😁).

Yup, try that then.

**Comment by TimBroddin** [comment]
I changed the stylesheet according to your suggestion @phcp

**Review by yansern — APPROVED** [review]
Looks good now.

image

**Pull request merged**